File: helidon-se/src/main/java/com/github/frederikpietzko/dto/CreateVisitDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Visit;

import java.time.OffsetDateTime;

public record CreateVisitDto(
  String description,
  CreatePetDto pet,
  OffsetDateTime dateTime
) {

  public Visit toVisit() {
    return new Visit(null, description, pet.toEntity(), dateTime);
  }
}

File: javalin/src/main/kotlin/com/github/frederikpietzko/dto/CreateVisitDto.kt
package com.github.frederikpietzko.dto

import java.time.OffsetDateTime

data class CreateVisitDto(
  val description: String,
  val pet: CreatePetDto,
  val dateTime: OffsetDateTime,
) {
  fun toModel() = com.github.frederikpietzko.model.Visit(
    id = null,
    description = description,
    pet = pet.toModel(),
    dateTime = dateTime,
  )
}
